pcbasic/basic/devices/ports.py

```diff
@@ -85,32 +88,34 @@ def available(self):
     def _parse_params(self, param):
         """Parse serial port connection parameters """
         max_param = 10
-        param_list = param.upper().split(b',')
+        if python_version[0]>'2': #For python3: convert byte string into str
+            param=param.decode('latin1')
+        param_list = param.upper().split(',')
```

Comment:
This is basically the right thing to do, I think - using `latin1` avoids the potential for decoding errors on malformed bytes strings that both `ascii` and `utf-8` could give you - BASIC knows nothing of UTF-8 and the user could put in byte values > 127 which will now be correctly caught by the `BAD_FILE_NAME` checks. 

One thing is that we shouldn't `import platform` and define `python_version` here as this is already done centrally in `pcbasic.compat`. So we can `from compat import PY2` and use `if PY2` here. Also we should leave spaces around operators such as `>`, `=` in agreement with the coding style of the rest of this code base (and PEP8 as well). 

But these are minor changes, I can make them after merging the commit if needed.